modules/honzon-distribution/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+// This file is part of Acala.
+
+// Copyright (C) 2020-2022 Acala Foundation.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! # Honzon Distribution Module
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+
+use codec::MaxEncodedLen;
+
+use frame_support::{pallet_prelude::*, transactional};
+use frame_system::pallet_prelude::*;
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{One, Saturating},
+	DispatchResult, FixedPointNumber, RuntimeDebug,
+};
+use sp_std::prelude::*;
+
+use module_support::Ratio;
+use nutsfinance_stable_asset::{traits::StableAsset as StableAssetT, StableAssetPoolId};
+use orml_traits::MultiCurrency;
+use primitives::{Amount, Balance, CurrencyId};
+
+mod mock;
+mod tests;
+pub mod weights;
+
+pub use module::*;
+pub use weights::WeightInfo;
+
+#[derive(Encode, Decode, Clone, RuntimeDebug, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub enum DistributionDestination<AccountId> {
+	StableAsset(DistributionToStableAsset<AccountId>),
+}
+
+/// Information needed when distribution to StableAsset.
+#[derive(Encode, Decode, Clone, RuntimeDebug, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub struct DistributionToStableAsset<AccountId> {
+	pub pool_id: StableAssetPoolId,
+	pub stable_token_index: u32,
+	pub stable_currency_id: CurrencyId,
```

Comment:
We could take this field and make it a config constant like `T::StableCurrencyId`. Do we want to use this pallet to mint something other than AUSD? if we do than make sense as is

Reply:
I think it'll be only AUSD. added.